src/config.rs

```diff
@@ -114,6 +118,84 @@ impl Default for Colors {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct Keymaps {
+    pub navigation: HashMap<KeyCode, Action>,
+}
+
+struct KeymapsVisitor;
+
+impl<'de> Deserialize<'de> for Keymaps {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        impl<'de> Visitor<'de> for KeymapsVisitor {
+            type Value = Keymaps;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                // TODO: Dont know what to write here
+                formatter.write_str("A map ?")
+            }
+
+            fn visit_map<A>(self, mut map: A) -> std::result::Result<Self::Value, A::Error>
+            where
+                A: serde::de::MapAccess<'de>,
+            {
+                let mut navigation = HashMap::new();
+
+                while let Some((section, section_values)) =
+                    map.next_entry::<String, HashMap<String, Vec<char>>>()?
```

Comment:
How do I expand this to allow the Vec to contain more than just characters ?

Reply:
I'm pretty sure you should just be able to replace the `Vec<char>` with `Vec<KeyCode>`, as `KeyCode` implements `Deserialize` already.